cypress/support/commands/commands.js

```diff
@@ -203,6 +203,70 @@ Cypress.Commands.add(
     }
 );
 
+/**
+ * Types in a sw-multi-select field all the specified values and checks if the content was correctly set.
+ * @memberOf Cypress.Chainable#
+ * @name typeMultiSelectAndCheckMultiple
+ * @function
+ * @param {String[]} values - Desired values of the element
+ */
+Cypress.Commands.add(
+    "typeMultiSelectAndCheckMultiple",
+    {
+        prevSubject: "element",
+    },
+    (subject, values) => {
+        // Request we want to wait for later
+        cy.server();
+        cy.route({
+            url: `${Cypress.env("apiPath")}/search/*`,
+            method: "post",
+        }).as("filteredResultCall");
+
+        cy.wrap(subject).should("be.visible");
+
+        // type in the search term if available
+        for (let i = 0; i < values.length; i += 1) {
```

Comment:
Could this loop cause timing issues in terms of the command queue? @klarstil If yes, we should wrap this loop in a Cypress-`then` command. See this [guide](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress.html#Mixing-Async-and-Sync-code) for more details about that.

Reply:
You are right. I think the `cy.wait` calls are not a good idea in a loop / synchronous code. I will try to refactor this.

Reply:
Okay I have read the docs about this topic now and I think even this loop should be completely fine. All it does is that it queues up commands to be executed and the order is still right. @klarstil correct me if I am wrong and need to change the loop somehow :thinking: .